lib/Conversion/TorchToMhlo/MhloLegalizeUtils.cpp

```diff
@@ -265,23 +262,97 @@ LogicalResult torchAlphaToMhloTensor(ConversionPatternRewriter &rewriter,
+
+  // Broadcast it
+  auto outType = tensorType.cloneWith(tensorType.getShape(), elementType);
+  if (tensorType.hasStaticShape()) {
+    return broadcastUnaryOperandStatic(rewriter, op, constOp, outType);
+  } else {
+    auto tensorShape = rewriter.create<shape::ShapeOfOp>(loc, tensor);
+    return rewriter.create<mhlo::DynamicBroadcastInDimOp>(
+        loc, outType, constOp, tensorShape, rewriter.getI64TensorAttr({}));
+  }
+}
+// template initialization
+template Value getConstTensorLike<float>(PatternRewriter &, Operation *, float,
+                                         Value, Type);
+template Value getConstTensorLike<int32_t>(PatternRewriter &, Operation *,
+                                           int32_t, Value, Type);
+
+// Broadcast the input tensor of 0 rank to a tensor with the same shape as the
+// given tensor. Supporting dynamic shape.
+Value broadcastZeroRankTensorToTensorLike(PatternRewriter &rewriter,
+                                          Operation *op, Value input,
+                                          Value tensor, Type elementType) {
+  Location loc = op->getLoc();
+  auto inputType = input.getType().cast<RankedTensorType>();
+  auto tensorType = tensor.getType().cast<RankedTensorType>();
+  if (inputType.getRank() != 0) {
+    op->emitError("Only zero ranked tensor is supported");
+  }
+  auto outType = tensorType.cloneWith(tensorType.getShape(), elementType);
+  input = promoteType(rewriter, input, outType);
+  if (tensorType.hasStaticShape()) {
+    return broadcastUnaryOperandStatic(rewriter, op, input, outType);
+  } else {
+    auto tensorShape = rewriter.create<shape::ShapeOfOp>(loc, tensor);
+    return rewriter.create<mhlo::DynamicBroadcastInDimOp>(
+        loc, outType, input, tensorShape, rewriter.getI64TensorAttr({}));
+  }
+}
+
+Value promoteAndBroadcast(ConversionPatternRewriter &rewriter, Value input,
+                          TensorType outType) {
+  Operation *op = input.getDefiningOp();
+
+  input = promoteType(rewriter, input, outType);
+  return broadcastUnaryOperandStatic(rewriter, op, input, outType);
+}
+
+Value promoteType(PatternRewriter &rewriter, Value input, TensorType outType) {
+  Operation *op = input.getDefiningOp();
+  auto inputType = input.getType().dyn_cast<mlir::TensorType>();
+
+  if (inputType.getElementType() != outType.getElementType()) {
+    TensorType promotedType =
+        inputType.cloneWith(inputType.getShape(), outType.getElementType());
+    return rewriter.create<mhlo::ConvertOp>(op->getLoc(), promotedType, input);
+  }
+  return input;
+}
+
+Value broadcastUnaryOperandStatic(PatternRewriter &rewriter, Operation *op,
+                                  Value input, TensorType outType) {
+
+  auto inputType = input.getType().dyn_cast<TensorType>();
+  if (!inputType) {
+    op->emitError("Only ranked tensor type input is supported!");
+  }
+  if (!inputType.hasStaticShape() || !outType.hasStaticShape()) {
```

Comment:
if `inputType` is null, may it corrupt at this line?

Reply:
Thanks, we forget some "return" sentences here.